### ISSUE TITLE
Candidate Stage Transition History

### ISSUE THREAD

**Issue by RishabhVerma** [navgurukul/chanakya#121]
## GET `/general/syncWithGSheet/addStageTransitionHistory`

Add the following endpoint. This endpoint will add the stage transition history of every candidate in a particular column.

A new column for every candidate needs to be added in the spreadsheet called `stageHistory`. This column will the transition history of each candidate. It might look something like this:

```
2 June, 2019: completedTest
3 June, 2019: testPassed
5 June, 2019: pendingAlgebraInterview
10 June, 2019: algebraInterviewFail
```

**Comment by pralhad88** [comment]
bhiya, If we use */general/syncWithGSheet/addPendingDataToSheet* this API end then this task will be done then why are you want to new API end to do this task?

**Comment by RishabhVerma** [comment]
Yes technically it can be done, but whenever we write something to the sheet it puts a lot of load on the sheet and it slows down for everyone who has the sheet open at that time. This happens due the real time nature of docs. That's why if you remember we split up the sync endpoint into two functional areas:

1. Adding Data
2. Syncing Stage Changes from the sheet to chanakya

The 2nd step just reads the sheet which doesn't slow down the sheet for the online users but writing does. That's why we run the 2nd operation every 5 minutes but do the 1st operation only once in 2 hours to prevent laggy performance. All write operations need to be done slowly. Having a separate endpoint for it will give us that control :)

Let me know if you have any doubts.
